client/state/data-layer/wpcom/me/notification-settings/index.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { NOTIFICATION_SETTINGS_REQUEST } from 'state/action-types';
+import { receiveNotificationSettings } from 'state/notification-settings/actions';
+import { errorNotice } from 'state/notices/actions';
+
+/**
+ * Dispatches a request to fetch the current user notification settings
+ *
+ * @param   {Function} dispatch Redux dispatcher
+ * @param   {Object}   action   Redux action
+ * @returns {Object}            dispatched http action
+ */
+export const requestNotificationSettings = ( { dispatch }, action ) => dispatch( http( {
+	apiVersion: '1.1',
+	method: 'GET',
+	path: '/me/notifications/settings',
```

Comment:
I wrote this a couple days ago, before the discussion on #14778. Updated on 7eb2aa510b02f2ef351bb34de37cdcf521a5c95a.